### PR TITLE
Support for SqlCmdVariables

### PR DESCRIPTION
**Pull request by Benjin**

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/SqlProjects/Contracts/SqlCmdVariables/AddSqlCmdVariable.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        /// <summary>
+        /// Name of the SQLCMD variable
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Default value of the SQLCMD variable
+        /// </summary>
+        public string DefaultVault { get; set; }
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/SqlProjects/Contracts/SqlCmdVariables/AddSqlCmdVariable.cs:24]: typo: Value

File: src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs [hunk 2]
```diff
@@ -43,10 +43,14 @@ public void InitializeService(ServiceHost serviceHost)
...
 
-            // SQL object script calls
+            // SQL object script functions
             serviceHost.SetRequestHandler(AddSqlObjectScriptRequest.Type, HandleAddSqlObjectScriptRequest, isParallelProcessingSupported: false);
             serviceHost.SetRequestHandler(DeleteSqlObjectScriptRequest.Type, HandleDeleteSqlObjectScriptRequest, isParallelProcessingSupported: false);
             serviceHost.SetRequestHandler(ExcludeSqlObjectScriptRequest.Type, HandleExcludeSqlObjectScriptRequest, isParallelProcessingSupported: false);
+
+            // SQLCMD variable functions
+            serviceHost.SetRequestHandler(AddSqlCmdVariableRequest.Type, HandleAddSqlCmdVariableRequest, isParallelProcessingSupported: false);
+            serviceHost.SetRequestHandler(DeleteSqlCmdVariableRequest.Type, HandleDeleteSqlCmdVariableRequest, isParallelProcessingSupported: false);
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs:53]: Will "EditSqlCmdVariable" also be supported?

File: test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SqlProjects/SqlProjectsServiceTests.cs [hunk 3]
```diff
@@ -153,6 +153,43 @@ await service.HandleDeleteSqlObjectScriptRequest(new SqlProjectScriptParams()
...
+                Value = "TestVarValue"
+            }, requestMock.Object);
+
+            Assert.IsTrue(requestMock.Result.Success, "Successful add request result expected");
+            Assert.AreEqual(1, service.Projects[projectUri].SqlCmdVariables.Count, "Number of SQLCMD variables after addition not as expected");
+            Assert.IsTrue(service.Projects[projectUri].SqlCmdVariables.Contains(variableName), $"List of SQLCMD variables expected to contain {variableName}");
+
+            // Validate deleting a SQLCMD variable
+            requestMock = new();
+            await service.HandleDeleteSqlCmdVariableRequest(new DeleteSqlCmdVariableParams()
```
Comment (kisantia) [inline test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SqlProjects/SqlProjectsServiceTests.cs:184]: please also add a test for trying to delete a sqlcmd variable that doesn't exist and adding a duplicate sqlcmd variable

File: src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs (hunk 2 above)
Comment (kisantia) [reply src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs:53]: sounds good!

File: src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs [hunk 4]
```diff
@@ -43,10 +43,14 @@ public void InitializeService(ServiceHost serviceHost)
             serviceHost.SetRequestHandler(CloseSqlProjectRequest.Type, HandleCloseSqlProjectRequest, isParallelProcessingSupported: true);
             serviceHost.SetRequestHandler(NewSqlProjectRequest.Type, HandleNewSqlProjectRequest, isParallelProcessingSupported: true);
 
-            // SQL object script calls
+            // SQL object script functions
             serviceHost.SetRequestHandler(AddSqlObjectScriptRequest.Type, HandleAddSqlObjectScriptRequest, isParallelProcessingSupported: false);
```
Comment (alanrenmsft) [inline src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs:47]: @caohai do you recall why parallel processing not supported for sql project requests?

Comment (caohai) [reply src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs:47]: @alanrenmsft I don't remember reviewing this file in my PR. I think this is a newly added service.

Comment (Benjin) [reply src/Microsoft.SqlTools.ServiceLayer/SqlProjects/SqlProjectsService.cs:47]: Correct, it's new.  The underlying SqlProject and MSBuild construction objects aren't explicitly threadsafe (could potentially hit some funky behavior with multiple operations hitting it at the same time), so I turned this off for operations that reach down that low.

**Comment by Benjin** [comment]
Abandoning this PR because github isn't updating it with the latest commits from the branch
